query/src/sql/planner/semantic/type_check.rs

```diff
@@ -586,6 +588,39 @@ impl<'a> TypeChecker<'a> {
                 .await?
             }
 
+            Expr::Coalesce { span, exprs } => {
+                // Rewrite COALESCE(expr1, expr2, ...) to IF(IS_NOT_NULL(expr1), expr1, expr2)   
+                let resolve_args = exprs.iter().rev().fold(
```

Comment:
We can filter exprs to remove `Literal::Null`

Reply:
I'll add the filter later.